### ISSUE TITLE
Sample names in results were all "null" when using FOFN

### ISSUE THREAD

**Issue by PHemarajata** [bactopia/bactopia#276]
Hi! I created the following FOFN:

```
sample 	runtype	r1	r2	extra
F59610	paired-end	/home/peerahemarajata/Documents/NGS_DATA/Vibrio/F59610_Vibrio-LAC-M04759-211201_S9_L001_R1_001.fastq.gz	/home/peerahemarajata/Documents/NGS_DATA/Vibrio/F59610_Vibrio-LAC-M04759-211201_S9_L001_R2_001.fastq.gz	
W31996A	paired-end	/home/peerahemarajata/Documents/NGS_DATA/Vibrio/W31996A_Vibrio-LAC-M04759-211201_S7_L001_R1_001.fastq.gz	/home/peerahemarajata/Documents/NGS_DATA/Vibrio/W31996A_Vibrio-LAC-M04759-211201_S7_L001_R2_001.fastq.gz	
W31996C	paired-end	/home/peerahemarajata/Documents/NGS_DATA/Vibrio/W31996C_Vibrio-LAC-M04759-211201_S8_L001_R1_001.fastq.gz	/home/peerahemarajata/Documents/NGS_DATA/Vibrio/W31996C_Vibrio-LAC-M04759-211201_S8_L001_R2_001.fastq.gz	
```
However, the workflow did not seem to recognize sample names in the FOFN and created all output files with "null" as sample names. I was wondering if you could kindly help point out what I did wrong. Thank you!

**Comment by rpetit3** [comment]
OK, this is now fix in https://github.com/bactopia/bactopia/commit/d9a4ba1dad7dae068742d181611c88517306ae46

It was super subtle, but sample column had a space after it (`sample ` instead of `sample`). To prevent this in the future, the columns will have whitespace removed at the start and end. Did you make this by had or use `bactopia prepare`? Either is fine, just curious if I need to fix something in `bactopia prepare`

But more importantly, parsing the FOFN will fail if it gets a `null` value for the sample name.

Thanks again for finding this!


---

If you like to give these changes a try, I have a latest build on my personal anaconda channel.

```
mamba create -n test-bactopia -c rpetit3 -c conda-forge -c bioconda bactopia
```

**Comment by PHemarajata** [comment]
Thank you! I decided to try the latest build but ran into this error.

 ```
Missing output file(s) *.{log,err} expected by process BACTOPIA:ARIBA_ANALYSIS (F59610_Vibrio-LAC-M04759-211201_S9_ - card)
```
I was wondering if this was because I didn't build new datasets within the test-bactopia environment. I just specified the datasets I already built for the latest stable release.

**Comment by rpetit3** [comment]
Haha catching me mid version! I've fixed this (https://github.com/bactopia/bactopia/commit/6b2827cd04f51fbe878c58d8f80d4a7579b62f72) and have rebuilt my dev install.

Thank you very much for finding this!

**Comment by PHemarajata** [comment]
Thank you! I think we are all good.

**Comment by rpetit3** [comment]
Great, if you run into any other issues let me know! More importantly, after you play around more any feed back would be greatly appreciated!

**Comment by rpetit3** [comment]
I've released v2.0.2 (https://github.com/bactopia/bactopia/releases/tag/v2.0.2) which includes a fix for this issue, but also many other changes! I'm going to go ahead and close this for now, please feel free to reopen!

If you have a moment, could you take a few minutes to fill out a [Bactopia Usage Survey](https://forms.gle/PTgUxEBwCsCdbQjr7)? It really helps me direct the direction of Bactopia's development! 

Thank you for spotting this!
